gui-js/apps/minsky-electron/src/app/managers/CommandsManager.ts

```diff
@@ -1428,4 +1421,94 @@ export class CommandsManager {
             filePath: filePath,
         });
     }
+
+    static async editHandleDescription(handleIndex: number) {
+      const description = await RestServiceManager.handleMinskyProcess({ command: `${commandsMapping.CANVAS_RAVEL_HANDLE_DESCRIPTION} ${handleIndex}` });
+
+      const window=WindowManager.createPopupWindowWithRouting({
+        title: `Handle Description`,
+        url: `#/headless/edit-handle-description?handleIndex=${handleIndex}&description=${description}`,
+        height: 90,
+        width: 300,
+      });
+      Object.defineProperty(window,'dontCloseOnReturn',{value: true,writable:false});
+    }
+
+    static async saveHandleDescription(payload: HandleDescriptionPayload) {
+      const args = JSON5.stringify([payload.handleIndex, payload.description]);
+      await RestServiceManager.handleMinskyProcess({ command: `${commandsMapping.CANVAS_RAVEL_HANDLE_SET_DESCRIPTION} ${args}`});
+    }
+
+    static async editHandleDimension(handleIndex: number) {
+      const type = await RestServiceManager.handleMinskyProcess({ command: commandsMapping.CANVAS_RAVEL_HANDLE_DIMENSION_TYPE });
+      const units = await RestServiceManager.handleMinskyProcess({ command: commandsMapping.CANVAS_RAVEL_HANDLE_DIMENSION_UNITS });
+
+      const window=WindowManager.createPopupWindowWithRouting({
+        title: `Handle Dimension`,
+        url: `#/headless/edit-handle-dimension?handleIndex=${handleIndex}&type=${type}&units=${units}`,
+        height: 180,
+        width: 300,
+      });
+      Object.defineProperty(window,'dontCloseOnReturn',{value: true,writable:false});
+    }
+
+    static async saveHandleDimension(payload: HandleDimensionPayload) {
+      const args = JSON5.stringify([payload.handleIndex, payload.type, payload.units]);
+      await RestServiceManager.handleMinskyProcess({ command: `${commandsMapping.CANVAS_RAVEL_HANDLE_SET_DIMENSION} ${args}` });   
+    }
+
+    static async pickSlices(handleIndex: number) {
+      const allSliceLabels = <string[]>await RestServiceManager.handleMinskyProcess({ command: commandsMapping.CANVAS_RAVEL_HANDLE_ALL_SLICE_LABELS });
```

Comment:
Is the <string[]> a type cast operator, like the "as string[]" suffix form?

Reply:
yes... although it's best to think of it as a type hint for the compiler... in runtime this does absolutely nothing. If I recall correctly, the handleMinskyProcess method returns an Unknown type, which you're pretty much forced to cast to something else to stop the compiler from complaining. One option we always have in typescript is to cast to <any>, which removes all type-based compilation checks. Can be convenient when you're in a hurry, but obviously a double-edged sword.